src/plugins/stake-plugins/generic/policyAdapters/TarotPoolAdaptor.ts

```diff
@@ -0,0 +1,491 @@
+            gasPrice,
+            gasLimit: '1000000',
+            customData: {
+              name: metadataName,
+              category: 'Transfer:Staking',
+              notes: `Deleverage ${metadataPoolAssetName} liquidity pool contract`
+            }
+          })
+      )
+
+      const allocations: QuoteAllocation[] = [
+        {
+          allocationType: 'unstake',
+          pluginId: policyConfig.stakeAssets[0].pluginId,
+          currencyCode: policyConfig.stakeAssets[0].currencyCode,
+          nativeAmount: bAmountAMin.toString()
+        },
+        {
+          allocationType: 'unstake',
+          pluginId: policyConfig.stakeAssets[1].pluginId,
+          currencyCode: policyConfig.stakeAssets[1].currencyCode,
+          nativeAmount: bAmountBMin.toString()
+        }
+      ]
+
+      return await prepareChangeQuote(walletSigner, txs, allocations)
+    },
+    fetchClaimQuote: (wallet: EdgeCurrencyWallet, requestAssetId: AssetId, nativeAmount: string) => {
+      throw new Error('fetchClaimQuote not implemented for TarotPoolAdapter')
+    },
+    fetchUnstakeExactQuote: async (wallet: EdgeCurrencyWallet, requestAssetId: AssetId, nativeAmount: string): Promise<ChangeQuote> => {
+      throw new Error('fetchUnstakeExactQuote not implemented for TarotPoolAdapter')
+    },
+    fetchStakePosition: async (wallet: EdgeCurrencyWallet): Promise<StakePosition> => {
+      const token0BorrowableContract = TarotBorrowable__factory.connect(token0BorrowableContractAddress, provider)
+      const token1BorrowableContract = TarotBorrowable__factory.connect(token1BorrowableContractAddress, provider)
+      const [token0, token1] = policyConfig.stakeAssets
+      const { walletAddress } = await workflowUtils(wallet)
+
+      const tarot = tarotUtils(adapterConfig, provider, walletAddress)
+
+      function getAllocation(token: AssetId, nativeAmount: string): PositionAllocation {
+        return {
+          pluginId: token.pluginId,
+          currencyCode: token.currencyCode,
+          allocationType: 'staked',
+          nativeAmount,
+          locktime: undefined
+        }
+      }
+
+      const allocations: PositionAllocation[] = []
+      const token0AccruedBalance = await tarot.getAccruedBalance(token0BorrowableContract, walletAddress)
+      const token0NativeAmount = token0AccruedBalance.toString()
+      allocations.push(getAllocation(token0, token0NativeAmount))
+
+      const token1AccruedBalance = await tarot.getAccruedBalance(token1BorrowableContract, walletAddress)
+      const token1NativeAmount = token1AccruedBalance.toString()
+      allocations.push(getAllocation(token1, token1NativeAmount))
+
+      //
+      // Actions available for the user:
+      //
+
+      const token0WalletBalance = wallet.balances[token0.currencyCode] ?? '0'
+      const token1WalletBalance = wallet.balances[token1.currencyCode] ?? '0'
```

Comment:
If the tokens aren't enabled, shouldn't they be enabled by the plugin, or the plugin shouldn't be accessible?